components/console_cmd_wifi/console_wifi.c

```diff
@@ -0,0 +1,289 @@
+    {.name = "leave", .operation = wifi_sta_leave_op,  .arg_cnt = 3, .start_index = 2, .help = "wifi sta leave: Station leaves the wifi network."},
+//    {.name = "stop", .operation = wifi_sta_stop_op,  .arg_cnt = 1, .start_index = 1, .help = "wifi"},
+};
+
+static void event_handler(void *arg, esp_event_base_t event_base,
+                          int32_t event_id, void *event_data)
+{
+    if (event_base == WIFI_EVENT && event_id == WIFI_EVENT_STA_START) {
+        xEventGroupSetBits(wifi_event_group, STA_STARTED_BIT);
+    } else if (event_base == WIFI_EVENT && event_id == WIFI_EVENT_STA_DISCONNECTED) {
+        esp_wifi_connect();
+        xEventGroupClearBits(wifi_event_group, CONNECTED_BIT);
+    } else if (event_base == IP_EVENT && event_id == IP_EVENT_STA_GOT_IP) {
+        xEventGroupSetBits(wifi_event_group, CONNECTED_BIT);
+    }
+}
+
+static esp_err_t wifi_help_op(wifi_op_t *self, int argc, char *argv[])
+{
+    int cmd_count = sizeof(cmd_list) / sizeof(cmd_list[0]);
+
+    for (int i = 0; i < cmd_count; i++) {
+        if ((cmd_list[i].help != NULL) && (strlen(cmd_list[i].help) != 0)) {
+            printf(" %s\n", cmd_list[i].help);
+        }
+    }
+
+    return ESP_OK;
+}
+
+uint8_t wifi_connection_status = 0;
+
+void wifi_init(void)
+{
+    static bool init_flag = false;
+    if (init_flag) {
+        return;
+    }
+    wifi_event_group = xEventGroupCreate();
+
+    esp_netif_t *sta_netif = esp_netif_create_default_wifi_sta();
+    assert(sta_netif);
+
+    wifi_init_config_t cfg = WIFI_INIT_CONFIG_DEFAULT();
+    ESP_ERROR_CHECK(esp_wifi_init(&cfg));
+
+    ESP_ERROR_CHECK(esp_event_handler_register(WIFI_EVENT, WIFI_EVENT_STA_START, &event_handler, NULL));
+    ESP_ERROR_CHECK(esp_event_handler_register(WIFI_EVENT, WIFI_EVENT_STA_DISCONNECTED, &event_handler, NULL));
+    ESP_ERROR_CHECK(esp_event_handler_register(IP_EVENT, IP_EVENT_STA_GOT_IP, &event_handler, NULL));
+
+    ESP_ERROR_CHECK(esp_wifi_set_storage(WIFI_STORAGE_RAM));
+    ESP_ERROR_CHECK(esp_wifi_set_mode(WIFI_MODE_STA));
+
+    ESP_ERROR_CHECK(esp_wifi_start());
+
+    int bits = xEventGroupWaitBits(wifi_event_group, STA_STARTED_BIT,
+                                   pdFALSE, pdTRUE, JOIN_TIMEOUT_MS / portTICK_PERIOD_MS);
+
+    if ((bits & STA_STARTED_BIT) == 0) {
+        ESP_LOGE(TAG, "Error: Wifi Connection timed out");
+        return;
+    }
+
+    init_flag = true;
+}
+
+/* Initialize Wi-Fi as sta and set scan method */
+static void wifi_scan(void)
+{
+    uint16_t number = DEFAULT_SCAN_LIST_SIZE;
+    wifi_ap_record_t ap_info[DEFAULT_SCAN_LIST_SIZE];
+    uint16_t ap_count = 0;
+
+    memset(ap_info, 0, sizeof(ap_info));
+
+    wifi_init();
+
+    esp_wifi_scan_start(NULL, true);
+    ESP_ERROR_CHECK(esp_wifi_scan_get_ap_records(&number, ap_info));
+    ESP_ERROR_CHECK(esp_wifi_scan_get_ap_num(&ap_count));
+
+    printf("\n");
```

Comment:
```suggestion
    ESP_LOGI(TAG, "\n");
```

Reply:
The new line was added to create a gap between the network list and the wifi initialization message the appears the first time `wifi show network` command is run. But now it doesn't seem necessary. Removing the print.